src/sentry/utils/kafka.py

```diff
@@ -64,13 +66,9 @@ def create_batching_kafka_consumer(topic_names, worker, **options):
 
     (cluster_name,) = cluster_names
 
-    bootstrap_servers = settings.KAFKA_CLUSTERS[cluster_name]["bootstrap.servers"]
-    if not isinstance(bootstrap_servers, (list, tuple)):
-        bootstrap_servers = bootstrap_servers.split(",")
```

Comment:
This means that the `bootstrap_servers` parameter (between legacy options) is supposed to be accepted whether it is a string with comma separated brokers or whether it is a sequence. Thus, sadly, you will have to allow for the same behavior in your kafka_config methods to keep this backward compatible. 
So the kafka consumer wants a string, if the config contains a sequence you need to join it and if it is a string I think you need to remove the commas.
I think you can get away in only supporting this for the legacy parameter, let's force common and custom parameters to be a sequence.

Reply:
In snuba we use comma separated strings, as this is also what librdkafka expects.
I will update common and custom to expect strings as well, not a sequence.